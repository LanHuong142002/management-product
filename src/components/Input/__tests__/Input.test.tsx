import { act, cleanup, fireEvent, render, screen } from '@testing-library/react';
import { Input, InputProps } from '@components';

describe('Testing Input component', () => {
  const handleChange = jest.fn();

  afterEach(() => {
    cleanup();
  });

  const defaultProps = {
    name: 'name',
    placeholder: 'Enter name...',
  } as InputProps;

  it('Should render Input component', () => {
    const { container } = render(<Input {...defaultProps} onChange={handleChange} />);

    expect(container).toMatchSnapshot();
  });

  it('Should call onChange event when change value input', () => {
    render(<Input {...defaultProps} onChange={handleChange} />);

    const input = screen.getByPlaceholderText<HTMLInputElement>('Enter name...');

    act(() => {
      fireEvent.change(input, {
        target: { value: 'full name' },
      });
    });

    expect(handleChange).toBeCalled();
    expect(input).toHaveValue('full name');
  });

  it('Should render component with title correctly', () => {
    const { getByText } = render(
      <Input {...defaultProps} title='Full Name' onChange={handleChange} />,
    );

    const text = getByText('Full Name');

    expect(text).toBeInTheDocument();
  });
});
