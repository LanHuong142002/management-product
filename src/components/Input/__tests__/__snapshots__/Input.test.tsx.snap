// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing Input component Should render Input component 1`] = `
<div>
  <input
    class="text-input text-input-default"
    name="name"
    placeholder="Enter name..."
    type="text"
    value=""
  />
</div>
`;
